infer/environment.h

```diff
@@ -214,6 +210,10 @@ class Environment {
 
     std::string toString(const core::GlobalState &gs, const cfg::CFG &cfg) const;
 
+    // NB: you can't call this function on vars in the first basic block since
+    // their type will be nullptr
+    const core::TypeAndOrigins &getTypeAndOrigin(core::Context ctx, cfg::LocalRef symbol) const;
```

Comment:
Was this private before because of the comment?

Reply:
Honestly, not sure.